src/TerraformEnvironmentStage.groovy

```diff
@@ -46,46 +45,13 @@ class TerraformEnvironmentStage implements Stage {
         Jenkinsfile.build(pipelineConfiguration())
     }
 
-    private Closure pipelineConfiguration() {
-        initCommand = TerraformInitCommand.instanceFor(environment)
-        planCommand = TerraformPlanCommand.instanceFor(environment)
-        applyCommand = TerraformApplyCommand.instanceFor(environment)
+    public void withStrategy(newStrategy) {
+        this.strategy = newStrategy
+    }
 
+    private Closure pipelineConfiguration() {
         applyPlugins()
-
-        def String environment = this.environment
-        return { ->
-            node(jenkinsfile.getNodeName()) {
-                deleteDir()
-                checkout(scm)
-
-                decorations.apply(ALL) {
-                    stage("${PLAN}-${environment}") {
-                        decorations.apply(PLAN) {
-                            sh initCommand.toString()
-                            sh planCommand.toString()
-                        }
-                    }
-
-                    decorations.apply("Around-${CONFIRM}") {
-                        stage("${CONFIRM}-${environment}") {
-                            decorations.apply(CONFIRM) {
-                                echo "Approved"
-                            }
-                        }
-                    }
-
-                    decorations.apply("Around-${APPLY}") {
-                        stage("${APPLY}-${environment}") {
-                            decorations.apply(APPLY) {
-                                sh initCommand.toString()
-                                sh applyCommand.toString()
-                            }
-                        }
-                    }
-                }
-            }
-        }
+        return strategy.createPipelineClosure(environment, decorations)
     }
```

Comment:
Same "If we only attack UseCase 1 in this PR (destroy-only), we no longer need these changes."